### PR TITLE
Resolve issues with automorphisms and Graphs and more general things

### PR DESCRIPTION
**Pull request by lkastner**
- Remove module declaration `Graphs`, as it is only causing confusion, see #1563 and many others
- `automorphisms(Graph)` are now `automorphism_group_generators(Graph)` (the latter already exists for other things)
- Move tests of `Graph` to `Combinatorics` section

### CODE REVIEW

**Comment by thofma** [comment]
OK, now it gets a bit more complicated. Since `automorphisms` was used in two different ways, we cannot use aliases to solve this (we can either make `automorphisms === automorphism_group_generators` or `automorphisms === automorphism_list` but not both).

So let's go down the deprecation route. I will push some changes here once I am done in Hecke.

**Comment by thofma** [comment]
@lkastner Are there other automorphism group things you want to do here?

**Comment by lkastner** [comment]
> @lkastner Are there other automorphism group things you want to do here?

I'll just add a paragraph to the style guide and fix the docs.

**Review by thofma — APPROVED** [review]

**Comment by fieker** [comment]
what about returning permutations or actual graph autmorphisms instead of Vec{Int}? Or should this be a seperate PR?

**Comment by lkastner** [comment]
> what about returning permutations or actual graph autmorphisms instead of Vec{Int}? Or should this be a seperate PR?

Yes, I want to make a separate PR for these things. This is already large enough.

**Comment by micjoswig** [comment]
Looks good to me.

File: docs/src/DeveloperDocumentation/styleguide.md [hunk 1]
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
...
 
-- Use `CamelCase` for types and `snake_case` for *everything* else. (Internal functions do not have to follow these rules.) Types (and their constructor) tend to be in `CamelCase`. However, please ALSO provide the constructor/ a constructor in underscore_case. As a
-user I do usually not know if something is a constructor or a function (nor do I
-want to).
-- Noteworthy difference to Julia base is that we do not have exceptions `is*` or `has*`.
+- Use `CamelCase` for types and `snake_case` for *everything* else. (Internal
+  functions do not have to follow these rules.) Types (and their constructor)
+  tend to be in `CamelCase`. However, please ALSO provide the constructor/ a
+  constructor in underscore_case. As a user I do usually not know if something
+  is a constructor or a function (nor do I want to).
```
Comment (fingolfin) [inline docs/src/DeveloperDocumentation/styleguide.md]:
```suggestion
  tend to be in `CamelCase`. However, please *also* provide the constructor (or a
  constructor) in `snake_case`. As a user one usually does not know if something
  is a constructor or a function.
```

File: docs/src/DeveloperDocumentation/styleguide.md [hunk 2]
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
...
+- Noteworthy difference to Julia base is that we do not have exceptions `is*`
+  or `has*`.
```
Comment (fingolfin) [inline docs/src/DeveloperDocumentation/styleguide.md]:
```suggestion
- Noteworthy difference to Julia base is that we do not have exceptions for
  `is*` or `has*` as prefix.
```

File: docs/src/DeveloperDocumentation/styleguide.md [hunk 3]
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
...
   It is `is_foo` instead of `isfoo` and `has_bar` instead of `hasbar`.
+  The main reason is to avoid awkward constructions like `isvery_ample`, while
+  also being consistent.
```
Comment (fingolfin) [inline docs/src/DeveloperDocumentation/styleguide.md:25]:
```suggestion
  also being consistent.
  For compatibility with standard Julia, while staying consistent internally,
  we also provide aliases (using `AbstractAlgebra.@alias`) for various standard
  Julia functions, e.g. `is_one` as alias for `isone`
```

File: docs/src/DeveloperDocumentation/styleguide.md [hunk 4]
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
...
 - For generic concepts choose generic names, based on general algebraic
   concepts, preferably not special names from your area of speciality.
 - Use Julia conventions where applicable.
```
Comment (fingolfin) [inline docs/src/DeveloperDocumentation/styleguide.md]:
```suggestion
- Use Julia conventions where applicable and when they don't contradict our
  own rules above.
```

**Comment by fingolfin** [comment]
@lkastner I was also editing the developer guidelines, and added some change suggestions here to make them match. Let me know if you prefer to merge this first, then I can also submit a separate PR with these changes later *shrug*

File: src/Deprecations.jl [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Deprecated in 0.10.*
+
+@deprecate automorphisms(x::Graph) automorphism_group_generators(x)
```
Comment (fingolfin) [inline src/Deprecations.jl:3]: What about `inner_automorphisms_group`, should it also be added here?

Comment (lkastner) [reply src/Deprecations.jl:3]: There is an alias for this.

**Comment by lkastner** [comment]
> @lkastner I was also editing the developer guidelines, and added some change suggestions here to make them match. Let me know if you prefer to merge this first, then I can also submit a separate PR with these changes later _shrug_

Thanks, I added your suggestions.

**Pull request merged**